### PR TITLE
Refactor/consolidate auth crd

### PR DESCRIPTION
**Pull request by maltesander**
# Description

consolidated auth config

fixes https://github.com/stackabletech/nifi-operator/issues/485

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by maltesander** [comment]
test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/nifi-operator-it-custom/40/

**Review by razvan — DISMISSED** [review]
Looks good. See the comment regarding the docs though.

File: rust/operator-binary/src/controller.rs [hunk 1]
```diff
@@ -220,10 +216,18 @@ pub enum Error {
         "failed to serialize [{JVM_SECURITY_PROPERTIES_FILE}] for {}",
         rolegroup
     ))]
-    JvmSecurityPoperties {
+    JvmSecurityProperties {
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/controller.rs:219]: no worries :)

File: docs/modules/nifi/pages/usage_guide/configuration-environment-overrides.adoc [hunk 2]
```diff
@@ -7,49 +7,23 @@ This will lead to cluster malfunction.
 
 == Configuration Overrides
 
-Apache NiFi runtime configuration is stored in several files listed below.  The `configOverrides` block enables you to customize parameters in these files.
+Apache NiFi runtime configuration is stored in the files bootstrap.conf and nifi.properties.
```
[the remark this answers is not in the document]
Comment (maltesander) [reply docs/modules/nifi/pages/usage_guide/configuration-environment-overrides.adoc:10]: Oh yeah i have a look thanks!

**Review by razvan — APPROVED** [review]
looks good

**Pull request merged**